### PR TITLE
consensus/pbft: refactor message types

### PR DESCRIPTION
**Pull request by alanchchen**

### CODE REVIEW

File: consensus/pbft/core/checkpoint.go [hunk 1]
```diff
@@ -22,39 +22,50 @@ import (
...
 }
 
-func (c *core) handleCheckpoint(cp *pbft.Checkpoint, src pbft.Validator) error {
+func (c *core) handleCheckpoint(msg *message, src pbft.Validator) error {
+	logger := c.logger.New("from", src.Address().Hex(), "state", c.state)
+
+	cp, ok := msg.Msg.(*pbft.Subject)
+	if !ok {
+		logger.Error("Invalid checkpoint message", "msg", msg)
+		return pbft.ErrIgnored
```
Comment (markya0616) [inline consensus/pbft/core/checkpoint.go]: return errFailedDecodeCheckpoint?

File: consensus/pbft/core/core.go [hunk 2]
```diff
@@ -97,23 +97,33 @@ type core struct {
 	snapshotsMu *sync.RWMutex
 }
 
-func (c *core) broadcast(code uint64, msg interface{}) {
+func (c *core) broadcast(msg *message) {
 	logger := c.logger.New("state", c.state)
```
Comment (markya0616) [inline consensus/pbft/core/core.go]: add address here?

File: consensus/pbft/core/snapshot.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ func newSnapshot(preprepare *pbft.Preprepare) *snapshot {
 		Preprepare:  preprepare,
 		Prepares:    pbft.NewMessageSet(preprepare.View, reflect.TypeOf(&pbft.Subject{})),
```
Comment (markya0616) [inline consensus/pbft/core/snapshot.go]: store `message` object?

File: consensus/pbft/core/snapshot.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func newSnapshot(preprepare *pbft.Preprepare) *snapshot {
...
 		Commits:     pbft.NewMessageSet(preprepare.View, reflect.TypeOf(&pbft.Subject{})),
```
Comment (markya0616) [inline consensus/pbft/core/snapshot.go]: store message object?

File: consensus/pbft/core/snapshot.go [hunk 5]
```diff
@@ -30,7 +30,7 @@ func newSnapshot(preprepare *pbft.Preprepare) *snapshot {
...
-		Checkpoints: pbft.NewMessageSet(preprepare.View, reflect.TypeOf(&pbft.Checkpoint{})),
```
Comment (markya0616) [inline consensus/pbft/core/snapshot.go]: store message object?

File: consensus/pbft/core/checkpoint.go (hunk 1 above)
Comment (alanchchen) [inline consensus/pbft/core/checkpoint.go]: yeah

File: consensus/pbft/core/snapshot.go (hunk 3 above)
Comment (alanchchen) [inline consensus/pbft/core/snapshot.go]: will be in next commit.

**Pull request merged**